src/IRMatch.h

```diff
@@ -1529,6 +1535,14 @@ HALIDE_ALWAYS_INLINE auto intrin(Call::IntrinsicOp intrinsic_op, Args... args) n
     return {intrinsic_op, pattern_arg(args)...};
 }
 
+template<typename A>
+auto abs(A &&a) noexcept -> Intrin<decltype(pattern_arg(a))> {
+    return {Call::abs, pattern_arg(a)};
```

Comment:
Here and in other helpers try adding `assert_is_lvalue_if_expr<A>();`

Reply:
Done in [9a5327c](https://github.com/halide/Halide/pull/6884/commits/9a5327c6243d8c53b1214150a2583eaceba44b98)